.github/workflows/gradle.yml

```
@@ -0,0 +1,31 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 17
+      uses: actions/setup-java@v2
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2
+    - name: Execute Gradle build
+      run: ./gradlew clean build
```

This should pass arguments instead of referencing the script itself
```suggestion
    with:
        arguments: build
```
https://github.com/gradle/gradle-build-action#use-the-action-to-setup-and-execute-gradle

It could also validate the wrapper prior to the build action:
```yml
- uses: gradle/wrapper-validation-action@v1
```
https://github.com/gradle/wrapper-validation-action#add-to-an-existing-workflow